src/internal/connector/graph_connector.go

```diff
@@ -73,7 +74,16 @@ func (gs graphService) ErrPolicy() bool {
 	return gs.failFast
 }
 
-func NewGraphConnector(ctx context.Context, acct account.Account) (*GraphConnector, error) {
+type resource int
+
+const (
+	UnknownResource resource = iota
+	AllResources
+	Users
+	Sites
+)
+
+func NewGraphConnector(ctx context.Context, acct account.Account, r resource) (*GraphConnector, error) {
```

Comment:
Can we also optimize so that if a specific user or site is specified - we don't enumerate *all* users/sites?

Reply:
Tracked as a separate issue here: https://github.com/alcionai/corso/issues/1547